crates/ruff_linter/src/rules/pep8_naming/rules/non_lowercase_variable_in_function.rs

```diff
@@ -53,6 +53,10 @@ impl Violation for NonLowercaseVariableInFunction {
 
 /// N806
 pub(crate) fn non_lowercase_variable_in_function(checker: &mut Checker, expr: &Expr, name: &str) {
+    if str::is_lowercase(name) {
+        return;
+    }
+
```

Comment:
What if the entire performance win is because of this unrelated lint rule change :smile: